### PR TITLE
Read extension info from pyproject.toml

### PR DESCRIPTION
**Pull request by lbertho-gpsw**
# Description
This PR add a function `create_extensions` that can be used in a `setup.py` like this:
```python
from setuptools import setup
from cython_setuptools import create_extensions

setup(ext_modules=create_extensions(__file__, cythonize=True))
```
Instead of reading a `setup.cfg` it gets the info from the pyproject.toml. Here's add example of what you can add to the pyproject.toml:
```toml
[cython_extensions.foo] # Extension will be named foo
sources = ["foo.pyx", "../src/foo.c"]
include_dirs = ["../src"]
```

I tried to add a lot of docstrings and type hints as possible

# Design choice
## Why not create/update a `setup` function?
I think creating a function with the smallest scope possible is better, this way we are also less dependent on setuptools

## Why not reuse/update existing code in vendors.py?
The existing code is really hard to follow, it was way simpler to start from scratch.
Also once all our project use a pyproject.toml, we will be able to entirely remove vendors.py 

## Why remove support for python 3.8 and python 3.9?
Normally we don't use python 3.8 and python 3.9 anymore. Also the dependency `pyserde` uses the type hints for the (de)serialization and it does only support the new python 3.10 type hint for `list` and `dict`

@mrobertseidowsky-gpsw I think you have the rights to do this, right now python3.8 and python3.9 are required for the PR but they are not built anymore so I think the PR requirements should be updated

# Behavior differences between `vendors.setup()` and `create_extensions` 
## Additional key value
With `vendors.setup()` in the setup.cfg it was possible to add any key value element and it was forwarded during the creation of the Extension. It was not added in `create_extensions` because it was not used. (except for adding "tags" but in vendors.py it is explicitly excluded form what is given to create the Extension...)
It this is needed in the future, it would probably a new array field with a named like this:
```toml
[cython_extensions.foo] 
sources = ["foo.pyx", "../src/foo.c"]
include_dirs = ["../src"]
additional_extensions_args = [{"my_key", 7}, {"other_key", "a value"}] # Name to be defined
```

## Parsing of pkg-config output
In vendors.py the output of pkg-config is tokenized once using `shlex.split()` then some specific args are extracted (`-I`, `-l`, `-L`) then other args are rejoined like `" ".join(args)` then `shlex.split()` is used again.

In `create_extensions` it uses `shlex.split()` once and we directly and that's it. The arguments are not parsed to extract specific flags as this is not needed (and if we would want to do it, then instead of parsing the ouput, using the pkg-config flags like `--cflags-only-I` and `--cflags-only-other` would be better than having to parse the output of pkg-config to find which flags is what)

The only impact this difference has is that if some flags that are not `-I`, `-l`, `-L` flags and they contains characters that must be escaped (like quotes or backslash) then with `create_extensions`  they must be escaped once, but with vendors.py they must be escaped several times.

### CODE REVIEW

File: cython_setuptools/common.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    """
+    Get the compiler flag for the corresponding version for the default compiler
+
+    Args:
+        version: C++ standard version (eg: 98, 11, 14, 17, 20, 23, 26)
+
+    Returns:
+        The compiler flag
+    """
+    return f"/std:c++{version}" if get_default_compiler() == "msvc" else f"-std=c++{version}"
```
Comment (mtreglia-gpsw) [inline cython_setuptools/common.py:22]:
```suggestion 
prefix = "/" if get_default_compiler() == "msvc" else "-"
return f"{prefix}std:c++{version}"
```

Comment (lbertho-gpsw) [reply cython_setuptools/common.py:22]: The separator between `std` and the version is also different. `:` for msvc and `=` otherwise
